File: guess.js
#!/usr/bin/env node

function game(numberToGuess) {
    var score = 0;
    console.log("Guess number I think of");
    return function guess(x) {
        score += 1;
        if (x == numberToGuess) {
            console.log("You WIN!\nScore:", score);
            process.exit();
        } else if (x > numberToGuess) {
            console.log("Smaller");
        } else if (x < numberToGuess) {
            console.log("Bigger");
        }
    };
}

var randToGuess = Math.floor(Math.random() *100);
process.stdin.on('data', game(randToGuess));

File: guess.lua
#!/usr/bin/env lua

function guess(answer)
    score = 0
    guess = -1
    while answer ~= guess do
        guess = tonumber(io.read())
        score = score + 1
        if guess > answer then
            print "Smaller"
        else
            print "Bigger"
        end
    end
    print "You WIN"
    return score
end

math.randomseed(os.time())
print("Guess number I think of")
print("Score: " .. guess(math.random(100)))
